sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionPump.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Messaging.EventHubs.Errors;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Constantly receives <see cref="EventData" /> from a single partition in the context of a given consumer
+    ///   group.  The received data is sent to an <see cref="IPartitionProcessor" /> to be processed.
+    /// </summary>
+    ///
+    internal class PartitionPump
+    {
+        /// <summary>
+        ///   A boolean value indicating whether this partition pump is currently running or not.
+        /// </summary>
+        ///
+        public bool IsRunning => RunningTask != null;
```

Comment:
What if your task is completed but not null?

Reply:
The only possible scenarios in which this task is completed are:

- When a stop is requested by the `EventProcessor`. The `Stop` method awaits the task to complete and makes `RunningTask` null.
- When there's a client exception. The `PartitionPump` calls its own `Stop` method, so the `RunningTask` is set to null as well.

I'll add an extra check to verify the task status just in case. I believe it makes more sense to verify it given the name of the member.

Reply:
Or, if your task experienced an unhandled exception.   You'd rethrow during `await` which happens before you set the `Task` to `null`.   In that case, you'd have a completed task that is not null.

I left some thoughts in `EventProcessor` that are also relevant here.   Should be a simple tweak to guard.